### ISSUE TITLE
[Brand FE] improve low-contrast Header links

### ISSUE THREAD

**Issue by andycochran** [HHS/simpler-grants-gov#3423]
### Summary

The links in the Header component are currently failing a11y contrast checks. The gray default state barely passes AA (fails AAA). But the green hover state fails AA. This is a result of adding our custom color palette. To remedy, we need to define explicit colors to use for the Header, as the defaults are too low-contrast. 

Unfortunately, this is not as simple as defining a theme setting since `$nav-link-color` [is hardcoded](https://github.com/uswds/uswds/blob/44e464c848edc08bc0b0b0ac80c50e68b9d2488e/packages/usa-nav/src/styles/_usa-nav.scss#L12) to use `base-dark`. 

[A bug has been filed for this](https://github.com/uswds/uswds/issues/6176) and the USWDS team has decided to add a theme setting. 

If we do not want to wait for this patch and want to change this now, it will require custom CSS to override the hardcoded default. 

### Acceptance criteria

- [ ] All states of Header links pass AA on https://webaim.org/resources/contrastchecker/

**Comment by andycochran** [comment]
@btabaska, this would be a great FOSS contribution issue. I'd recommend that we contribute back to USWDS and make `$nav-link-color` and `$nav-link-hover-color` theme settings

**Comment by margaretspring** [comment]
@andycochran Can this be closed?

**Comment by andycochran** [comment]
This bug still exists. We should probably keep it open until USWDS releases a fix.

**Comment by andycochran** [comment]
Update: Platform discussed this (well, me & @doug-s-nava) and decided that the resolution here is to fix this on our end with a little custom CSS to override the hardcoded default. We can contribute back, but let's not expect a USWDS patch for this soon.

**Comment by andycochran** [comment]
We need to specify the colors that should be used for all states and viewport sizes. Removing the FOSS labels for now, until we refine this ticket with design decisions and a mockup.

**Comment by andycochran** [comment]
Fixed this in #9395. Made some judgement calls about the colors, but I'll review w/ the Design Team on Monday.
